src/replayer_mismatch/crypto.py

```diff
@@ -0,0 +1,16 @@
+import base64
+from Crypto.Cipher import AES
+
+
+def decrypted_data_key(kms_client, encrypted_key):
+    cipher = base64.urlsafe_b64decode(encrypted_key)
+    decrypted_key_response = kms_client.decrypt(CiphertextBlob=cipher)
+    return decrypted_key_response["Plaintext"]
+
+
+def decrypted_take_home_pay(decrypted_key, encrypted_take_home_pay):
+    iv = encrypted_take_home_pay[:16]
+    encrypted = encrypted_take_home_pay[16:]
+    aes = AES.new(decrypted_key, AES.MODE_GCM, nonce=base64.urlsafe_b64decode(iv))
+    raw = base64.urlsafe_b64decode(encrypted)
+    return aes.decrypt_and_verify(raw[:-16], raw[-16:]).decode("ASCII")
```

Comment:
Is there a reason that you are using `[-16:]` and `[:-16]` on line 16 and `[16:]` and `[:16]` on line 12 and 13? Would be easier to read if they were the same format, but I can't work out if you are just splitting them in half, or there is another reason.

Reply:
Ye, I understand what the functionality of the code is, I think my question (albeit badly worded) was more "is the string 32 chars long and you are splitting it down the middle or, are you taking a 16 char subset of a non 32 char string?"

Reply:
Yep the latter, both strings are over 32 characters.